hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/security/x509/certificate/client/DefaultCertificateClient.java

```diff
@@ -110,28 +120,35 @@ public abstract class DefaultCertificateClient implements CertificateClient {
   private long localCrlId;
   private String component;
   private List<String> pemEncodedCACerts = null;
-  private final Lock lock;
   private KeyStoresFactory serverKeyStoresFactory;
   private KeyStoresFactory clientKeyStoresFactory;
+  private AtomicBoolean isRenewing = new AtomicBoolean(false);
+  private AtomicBoolean isRenewed = new AtomicBoolean(false);
 
-  DefaultCertificateClient(SecurityConfig securityConfig, Logger log,
+  DefaultCertificateClient(OzoneConfiguration conf, Logger log,
```

Comment:
I am not sure what is the advantage of changing here to OzoneConfiguration from SecurityConfig, can you share some reasoning for this change please?

Reply:
getCSRBuilder and create scm secure client requires OzoneConfiguration as paremeter.  These two functions are refactored from component service to CertificateClient. So previously CertificateClient only requires SecurityConfig, now OzoneConfiguration will be better.